libass/ass_fontselect.h

```diff
@@ -148,13 +148,15 @@ typedef void    (*SubstituteFontFunc)(void *priv, const char *name,
  * \param lib ASS_Library instance
  * \param family original font family name (try matching a similar font) (never NULL)
  * \param codepoint Unicode codepoint (UTF-32)
+ * \param locale UTS#35 locale to use for the lookup, or NULL for system default
```

Comment:
Minor: the mention of UTS#35 seems awfully specific. Do we know that our font provider backends specifically expect _Unicode locale identifiers?_ Does HarfBuzz (which is what the tracks’s `Language` is currently used for) expect the same? And at any rate, we only currently read two bytes of the `Language` header (but technically a consumer may put arbitrary data in through the API), although IIRC we’ve had requests to fix—er, expand this.

Reply:
UTS#35 is what CoreText expects; Fontconfig takes RFC3066 in a way that makes both forms effectively equivalent. We could call it "RFC 3066" here if we wanted since that's technically slightly stricter.

Reply:
("BCP 47" is another common name for these)

Reply:
Fontconfig according to its documentation accepts “the form Ll-Tt where Ll is a two or three letter language from ISO 639 and Tt is a territory from ISO 3166”.

Notably, this differs from BCP47 in that the latter can have a “script” identifier before the region identifier which Fontconfig doesn't expect or understand and in that BCP47 requires the shortest ISO 369 language code to be used while Fontconfig doesn't.

If we take BCP47 then without maintaining our own database of identifiers, we can iinm pass it as is to HarfBuzz, check if the first part consists of two letters and only pass it to libunibreak if that’s the case. It appears like script tags are always 4letter while region tags can have 2 or 3 letters. We could use this to sanitise the identifier to a form accepted by Fontconfig. Or can `FcLangNormalize` already handle this?

Nit: IETF BCP 47 actually contains RFC5646, which obsoleted the RFC obsoleting RFC3066.